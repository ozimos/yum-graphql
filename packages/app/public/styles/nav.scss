nav.flexbox{
    background-color: #fafafa;
    a, button{
        font-family: Roboto, Monserrat, 'Segoe UI', Tahoma, Verdana, sans-serif;
        color: #2f80ed; 
        line-height: normal;                    
        &.active{
            font-weight: bold;
            font-family: Arvo, Montserrat, cursive, 'sans-serif';
            padding-bottom: 5px
          } 
      }
      button{
          border: none;
          cursor: pointer;
          background-color: white
      }
  }
