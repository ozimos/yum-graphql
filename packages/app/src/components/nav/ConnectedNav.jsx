import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { userActions } from '../../redux/actions';
import '../../../public/styles/nav.scss';

class Nav extends React.Component {

   logoutHandler = () => {
     this.props.dispatch(userActions.logout());
     this.props.dispatch(push('/login'));
   }

   render() {
     const { user, authenticated } = this.props;

     return (
       <div className="navbar-fixed">
         <nav className="flexbox">

           <h3 className="shrink heading">
             <Link to="/">
        Book A Meal
             </Link>
           </h3>

           <div className="flexbox">

             {user.isCaterer &&
             <NavLink
               activeClassName="active"
               className="nav-item long_string"
               to="/dashboard"
             >
        DashBoard
             </NavLink >}

             {user.isCaterer &&
             <NavLink
               activeClassName="active"
               className="nav-item"
               to="/meals"
             >
          Meals
             </NavLink >}

             {user.isCaterer &&
             <NavLink
               activeClassName="active"
               className="nav-item"
               to="/menu"
             >
          Menu
             </NavLink>}

             {!user.isCaterer &&
             <NavLink
               activeClassName="active"
               className="nav-item"
               to="/orders"
             >
          Meal Booking
             </NavLink>}

             {authenticated &&
             <button
               onClick={this.logoutHandler}
             >
          Log Out
             </button>}

             {!authenticated &&
             <Link to="/login">
          Log In
             </Link>}

           </div>
         </nav>
       </div>
     );

   }
}

Nav.defaultProps = {
  user: { isCaterer: false }
};

Nav.propTypes = {
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
  }),
  authenticated: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.loginReducer.user,
  authenticated: state.loginReducer.authenticated,
});

export { Nav };

export default connect(mapStateToProps)(Nav);
