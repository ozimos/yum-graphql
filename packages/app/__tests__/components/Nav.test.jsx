/* global React:false, shallow:false toJson:false */

import { Nav } from '../../src/components/nav/ConnectedNav';

const props = {
  user: { isCaterer: true },
  authenticated: true,
  dispatch: jest.fn()
};

const propsCustomer = {
  user: { isCaterer: false },
  authenticated: true,
  dispatch: jest.fn()
};

describe('Nav Component', () => {
  const setup = () => shallow(<Nav {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect((wrapper)
      .exists(<nav className="flexbox" />)).toBe(true);
  });

  it('renders correctly', () => {

    const propsUnAuthenticated = {
      user: { isCaterer: false },
      authenticated: false,
      dispatch: jest.fn()
    };
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();

    const wrapperCustomer = shallow(<Nav {...propsCustomer} />);
    expect(toJson(wrapperCustomer)).toMatchSnapshot();

    const wrapperUnAuthenticated = shallow(<Nav {...propsUnAuthenticated} />);
    expect(toJson(wrapperUnAuthenticated)).toMatchSnapshot();

  });

  it('should call `logoutHandler`', () => {
    const wrapper = setup();
    wrapper.find('button').simulate('click');

    const logoutHandlerSpy = jest.spyOn(
      wrapper.instance(),
      'logoutHandler'
    );
    wrapper.instance().logoutHandler();
    expect(logoutHandlerSpy).toHaveBeenCalled();
  });
});

