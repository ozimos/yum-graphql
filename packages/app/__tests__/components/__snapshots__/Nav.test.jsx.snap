// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component renders correctly 1`] = `
<div
  className="navbar-fixed"
>
  <nav
    className="flexbox"
  >
    <h3
      className="shrink heading"
    >
      <Link
        replace={false}
        to="/"
      >
        Book A Meal
      </Link>
    </h3>
    <div
      className="flexbox"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item long_string"
        to="/dashboard"
      >
        DashBoard
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item"
        to="/meals"
      >
        Meals
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item"
        to="/menu"
      >
        Menu
      </NavLink>
      <button
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </nav>
</div>
`;

exports[`Nav Component renders correctly 2`] = `
<div
  className="navbar-fixed"
>
  <nav
    className="flexbox"
  >
    <h3
      className="shrink heading"
    >
      <Link
        replace={false}
        to="/"
      >
        Book A Meal
      </Link>
    </h3>
    <div
      className="flexbox"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item"
        to="/orders"
      >
        Meal Booking
      </NavLink>
      <button
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </nav>
</div>
`;

exports[`Nav Component renders correctly 3`] = `
<div
  className="navbar-fixed"
>
  <nav
    className="flexbox"
  >
    <h3
      className="shrink heading"
    >
      <Link
        replace={false}
        to="/"
      >
        Book A Meal
      </Link>
    </h3>
    <div
      className="flexbox"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item"
        to="/orders"
      >
        Meal Booking
      </NavLink>
      <Link
        replace={false}
        to="/login"
      >
        Log In
      </Link>
    </div>
  </nav>
</div>
`;
